controllers/runner_controller.go

```diff
@@ -927,3 +944,82 @@ func workVolumeMountPresent(items []corev1.VolumeMount) (bool, int) {
 	}
 	return false, 0
 }
+
+func applyWorkVolumeClaimTemplate(runner *v1alpha1.Runner) error {
+	isPresent, _ := workVolumeMountPresent(runner.Spec.VolumeMounts)
+	if isPresent {
+		return errors.New("volumeMount with name \"work\" should not exist if workVolumeClaimTemplate is specified")
+	}
+
+	isPresent, _ = workVolumePresent(runner.Spec.Volumes)
+	if isPresent {
+		return errors.New("volume with name \"work\" should not exist if workVolumeClaimTemplate is specified")
+	}
+
+	runner.Spec.VolumeMounts = append(runner.Spec.VolumeMounts, corev1.VolumeMount{
+		MountPath: "/runner/_work",
+		Name:      "work",
+	})
+
+	runner.Spec.Volumes = append(runner.Spec.Volumes, corev1.Volume{
+		Name: "work",
+		VolumeSource: corev1.VolumeSource{
+			Ephemeral: &corev1.EphemeralVolumeSource{
+				VolumeClaimTemplate: &corev1.PersistentVolumeClaimTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: runner.ObjectMeta.Name + "-work",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes:      runner.Spec.WorkVolumeClaimTemplate.AccessModes,
+						StorageClassName: &runner.Spec.WorkVolumeClaimTemplate.StorageClassName,
+						Resources:        runner.Spec.WorkVolumeClaimTemplate.Resources,
+					},
+				},
+			},
+		},
+	})
+	return nil
+}
+
+// isRequireSameNode specifies for the runner in kubernetes mode wether it should
+// schedule jobs to the same node where the runner is
+func isRequireSameNode(pod *corev1.Pod) (bool, error) {
+	isPresent, index := workVolumePresent(pod.Spec.Volumes)
+	if !isPresent {
+		return true, nil
+	}
+
+	if pod.Spec.Volumes[index].Ephemeral == nil || pod.Spec.Volumes[index].Ephemeral.VolumeClaimTemplate == nil {
```

Comment:
I don't know how to proceed if user wants to create volumes  without using `WorkVolumeClaimTemplate`. We are turning it into ephemeral, but what if they use volume definition and not `workVolumeClaimTemplate` field?

Reply:
`WorkVolumeClaimTemplate` should we require this if containerMode is kubernetes?